### ISSUE TITLE
🦔  Why are there candidates "Screened in" in previously cleared pools?

### ISSUE THREAD

**Issue by gobyrne** [GCTC-NTGC/gc-digital-talent#11744]
## Question

Recruitment team reporting that pools where they had processed absolutely everyone (as in, all applicants had been screened out, removed, or placed,_ somehow have new people marked as "screened in" a year later.

How might this have happened?

## 🕵️ Details

This has been reported by David and Sophie. Normally David reviews new applications to a pool (candidates with NEW APPLICATION status) and screens them in or out. Sophie then looks at the candidate table for a pool, filters by SCREENED_IN status, and continues with other assessments until all candidates are QUALIFIED or DISQUALIFIED.

We have as examples of the problem, three IT-ongoing pools from last year. You can find these pools by filtering by
- IT-02/IT-03/IT-04
- Project Portfolio Management stream
- IT Jobs (ongoing) publishing group

D and S report that as of the end of last summer, there were no candidates in these pools who were still NEW APPLICATION status or SCREENED_IN status. Then at some point in Oct 2024, they looked at these pools and noticed there were some candidates with a SCREENED_IN status in these pools. Also, they don't remember screening-in these names at all.

## 📋 Steps to Reproduce

1. Filter pools in production by
- IT-02/IT-03/IT-04
- Project Portfolio Management stream
- IT Jobs (ongoing) publishing group
2. Filter candidates by SCREENED_IN status
3. See that there are candidates left

## 🙋‍♀️ Proposed Solution

We want to understand why these candidates are here. Did they gain the SCREENED_IN status in an unexpected way? Were they hidden from the table for some amount of time?

Some ideas:
- check the database for submitted_at and updated_at timestamps. Were they last updated in an expected or unexpected time window?
- Check the activity-log for these candidates. Does it show when they become screened in, or anything unexpected?
- Is there anything in common between these candidates or their user profiles? Its possible there was a bug in the table at some point that caused them not to appear, or to be filtered out unintentially
- Is it possible they had a REMOVED status at some point (candidates suspending themselves?) which was later reverted?

**Comment by brindasasi** [comment]
Is it possible to specify the process name in prod?

**Comment by brindasasi** [comment]
are you workign now

**Comment by brindasasi** [comment]
I took a specific pool IT-02 which is published on 2023-04-14 and closed on 2023-08-16
The applicants who are all show up as mysterious candidates currently are applied within those specific timeframe and got assigned to an expiry date same as pool expiry date 2023-08-16 🙈

There is an evidence that one of these candidates got screened in around December 8 2023. 
```
{
    "attributes": {
        "updated_at": "2023-12-08T19:01:46.000000Z",
        "expiry_date": "2025-08-16T00:00:00.000000Z",
        "pool_candidate_status": "SCREENED_IN",
        "notes": "bug - expired status. Screened in on December 8, 2023",
        "status_weight": 50
    },
    "old": {
        "updated_at": "2023-08-11T20:28:20.000000Z",
        "expiry_date": "2023-08-16T00:00:00.000000Z",
        "pool_candidate_status": "NEW_APPLICATION",
        "notes": null,
        "status_weight": 30
    }
}
```

and the note says , the candidate got screened in even though the status looks as expired which is not 

We noticed this bug sometime in feb 2024 and ran a script to wipe out all expiry date on pool_candidates table

**Comment by brindasasi** [comment]
As I spoke with David & Sophie , they don't remember the timelines and we don't have any logs before March 2024.

So the conclusion would be , the candidates were screened in around Dec.8 and unfortunately having expiry date assigned due to the bug we had at that time. It made these records hidden during the later session of assessment and they came back after we ran the script to fix the bug on production. 

I've confirmed this bug doesn't exist in production anymore and none of the screened-in candidates are hidden in production. 

I will send broader communication to the recruitment team on what happened.
